tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/object/ObjectTest.java

```diff
@@ -921,4 +921,11 @@ private Object[][] checkInObjectFieldInitializerTests() {
                 {"testCheckInObjectFieldInitializer6"}
         };
     }
+
+    @Test
+    public void testNonPublicSymbolsWarningInClientDecl() {
+        CompileResult result = BCompileUtil.compile("test-src/object/client_object_decl_with_non_public_symbols.bal");
+        Assert.assertEquals(result.getWarnCount(), 1);
+        BAssertUtil.validateWarning(result, 0, "attempt to expose non-public symbol 'Foo'", 18, 5);
```

Comment:
Let's use a variable and assert the diagnostic count against it. This is to make sure that we continue to validate any and all diagnostics here.